library/kotlin/src/io/envoyproxy/envoymobile/ResponseHandler.kt

```diff
@@ -12,45 +12,49 @@ import java.util.concurrent.Executor;
 class ResponseHandler(val executor: Executor) {
 
   class EnvoyHTTPCallbacksAdapter(
-      private val responseHandler: ResponseHandler
+      private val executor: Executor
   ) : EnvoyHTTPCallbacks {
 
-    override fun getExecutor(): Executor = responseHandler.executor
+    private var onHeadersClosure: (headers: Map<String, List<String>>, statusCode: Int, endStream: Boolean) -> Unit = { _, _, _ -> Unit }
+
+    internal var onDataClosure: (byteBuffer: ByteBuffer, endStream: Boolean) -> Unit = { _, _ -> Unit }
```

Comment:
Would be nice if these could be lazily allocated

Reply:
I can just do a null check if that is better, this just ensures we have something to call into regardless of the setting state